custom/nic_compliance/static/nic_compliance/js/encoder.js

```diff
@@ -0,0 +1,53 @@
+hqDefine('nic_compliance/js/encoder.js', function () {
+    function HexParsr() {
+        function paddingStr() {
+            var s = Math.random().toString(36).slice(2, 8);
+            return s.length === 6 ? s : paddingStr();
+        }
+
+        // private property
+        var _paddingLeft = "sha256$" + paddingStr();
+        var _paddingRight = paddingStr() + "=";
+
+        var b64EncodeUnicode = function(str){
+            return encodeURIComponent(str).replace(/%([0-9A-F]{2})/g, function(match, p1) {
+                return String.fromCharCode('0x' + p1);
+            });
+        };
+
+        this.addPadding = function(secret_password) {
+            return _paddingLeft + secret_password + _paddingRight;
+        };
+
+        this.encode = function(password) {
+            password = password.trim();
```

Comment:
It doesn't seem like a good practice to trim characters off the password entered by the user. Making the password shorter makes it weaker and easier to guess. I still think we should not do this.